mlir/lib/Analysis/Presburger/Matrix.cpp

```diff
@@ -29,6 +29,24 @@ Matrix<T>::Matrix(unsigned rows, unsigned columns, unsigned reservedRows,
   data.reserve(std::max(nRows, reservedRows) * nReservedColumns);
 }
 
+template <typename T>
+bool Matrix<T>::operator==(const Matrix<T> &m) const {
+  if (nRows != m.getNumRows())
+    return false;
```

Comment:
leave a comment mentioning that the default doesn't work because of reserved columns